pyrokinetics/kinetics/Kinetics.py

```diff
@@ -48,19 +50,32 @@ def __init__(
         self,
         kinetics_file: PathLike,
         kinetics_type: Optional[str] = None,
+        eq_file: Optional[PathLike] = None,
         **kwargs,
     ):
         self.kinetics_file = Path(kinetics_file)
 
         if kinetics_type is not None:
-            reader = kinetics_readers[kinetics_type]
-            self.kinetics_type = kinetics_type
+            if kinetics_type == "pFile":  # if pFile.
+                reader = kinetics_readers[kinetics_type]
+                self.kinetics_type = kinetics_type
+                self.eq_file = Path(eq_file)
+            else:
+                reader = kinetics_readers[kinetics_type]
+                self.kinetics_type = kinetics_type
         else:
             # Infer kinetics type from file
             reader = kinetics_readers[kinetics_file]
             self.kinetics_type = reader.file_type
 
-        self.species_data = CleverDict(reader(kinetics_file, **kwargs))
+        if kinetics_type == "pFile":
+            self.species_data = CleverDict(
+                reader(kinetics_file, eq_file, **kwargs)
+            )  # Use reader __call__
+        else:
+            self.species_data = CleverDict(
+                reader(kinetics_file, **kwargs)
+            )  # Use reader __call__
```

Comment:
Instead of special-casing the pFile reader, it's probably better to make all the readers take an (optional) equilibrium file. Then the `if else` above at L59 isn't needed either

Reply:
I'm working on this now, and I found a cleaner solution is just to pass `eq_file` via `**kwargs`, and raise an error if it isn't present for the pFile reader.